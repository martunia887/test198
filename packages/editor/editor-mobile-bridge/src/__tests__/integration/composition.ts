import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

import {
  editor,
  editable,
  getDocFromElement,
  skipBrowsers as skip,
} from './_utils';

BrowserTestCase(
  `composition.ts: Support state update between composition events`,
  { skip: skip.concat('safari'), mobile: true },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, "soon we'll test composition here");
    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
