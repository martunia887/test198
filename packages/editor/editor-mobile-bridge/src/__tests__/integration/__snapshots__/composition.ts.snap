// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition.ts: Support state update between composition events 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "soon weâ€™ll test composition here",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
